### PR TITLE
Online Report

### PR DESCRIPTION
**Pull request by nerya10**

### CODE REVIEW

File: cli/flows-cli.sh [hunk 1]
```diff
@@ -265,12 +267,23 @@ else
   echo "[ ] Not a tty - will not run interactive"
 fi
 
+
+if [ ${PIIANO_CS_SCAN_VIEW} = "online" ] ; then
```
Comment (david-piiano) [inline cli/flows-cli.sh:271]: I suggest to move into a function above and also must check the CURL output and not assume it works

File: cli/flows-cli.sh [hunk 2]
```diff
@@ -28,6 +29,7 @@ FLOWS_PORT=3000
 PORT_START_RANGE=${FLOWS_PORT}
 PORT_END_RANGE=$(( ${PORT_START_RANGE} + 128 ))
 AWS_CLI_DOCKER=amazon/aws-cli:2.13.15
+NETWORK_PARAM=${NETWORK_PARAM:-""}
```
Comment (david-piiano) [inline cli/flows-cli.sh:32]: Please remind me what are you going to pass here?

File: cli/flows-cli.sh [hunk 3]
```diff
@@ -293,6 +307,11 @@ else
       ${PIIANO_CS_ENGINE_IMAGE} ${EXTRA_TEST_PARAMS[@]:-}
 fi
 
+if [ ${PIIANO_CS_SCAN_VIEW} = "online" ] ; then
+  echo "Flows viewer is ready at: ${VIEWER_URL}/${PIIANO_CS_SCAN_ID_EXTERNAL}"
```
Comment (david-piiano) [inline cli/flows-cli.sh:311]:
```suggestion
  echo "Your report is ready at: ${VIEWER_URL}/${PIIANO_CS_SCAN_ID_EXTERNAL}"
```

File: cli/flows-cli.sh [hunk 4]
```diff
@@ -18,6 +18,7 @@ PIIANO_CS_ENGINE_IMAGE=piiano/code-scanner:offline-${ENGINE_VERSION}
 PIIANO_CS_VIEWER_IMAGE=piiano/flows-viewer:${VIEWER_VERSION}
 PIIANO_CS_TAINT_ANALYZER_LOG_LEVEL=${PIIANO_CS_TAINT_ANALYZER_LOG_LEVEL:-'--verbosity=progress'}
 FLOWS_SKIP_ENGINE=${FLOWS_SKIP_ENGINE:-false}
+PIIANO_CS_SCAN_VIEW=${PIIANO_CS_SCAN_VIEW:-offline}
```
Comment (david-piiano) [inline cli/flows-cli.sh:21]: It is becoming a bit of spaghetti - how about we unify SKIP_VIEWER and CS_SCAN_VIEW with the following:
`PIIANO_CS_VIEWER_MODE=online|local|none`
Default for now is `local`

File: cli/flows-cli.sh (hunk 2 above)
Comment (nerya10) [reply cli/flows-cli.sh:32]: --network=<network name>

in order to connect minio container to engine container

**Review by david-piiano — APPROVED** [review]

File: cli/flows-cli.sh [hunk 5]
```diff
@@ -139,6 +140,17 @@ initial_cleanup()
     fi
 }
 
+get_external_id() {
+  BACKEND_TOKEN="${BACKEND_TOKEN:-$ACCESS_TOKEN}"
+  BACKEND_URL="${BACKEND_URL:-https://scanner.piiano.io/api/app/scans}"
+  REPOSITORY_URL=$(grep url "${PATH_TO_SOURCE_CODE}/.git/config" | awk '{print $3}')
+  SCAN_NAME="${SCAN_NAME:-no_scan_name}"
+  PIIANO_CS_SCAN_ID_EXTERNAL=$(curl --silent --fail --location -X POST -H 'Content-Type: application/json' -H "Authorization: Bearer ${BACKEND_TOKEN}" -d "{\"name\": \"${SCAN_NAME}\",\"subDir\": \"${PIIANO_CS_SUB_DIR}\",\"repositoryUrl\": \"${REPOSITORY_URL}\",\"runningMode\": \"offline\"}" ${BACKEND_URL} | jq -r '.uid')
+  echo "scan id ${PIIANO_CS_SCAN_ID_EXTERNAL}"
```
Comment (david-piiano) [inline cli/flows-cli.sh:149]:
```suggestion
  echo "[ ] New scan id ${PIIANO_CS_SCAN_ID_EXTERNAL}"
```

File: cli/flows-cli.sh [hunk 6]
```diff
@@ -139,6 +140,17 @@ initial_cleanup()
     fi
 }
 
+get_external_id() {
+  BACKEND_TOKEN="${BACKEND_TOKEN:-$ACCESS_TOKEN}"
+  BACKEND_URL="${BACKEND_URL:-https://scanner.piiano.io/api/app/scans}"
+  REPOSITORY_URL=$(grep url "${PATH_TO_SOURCE_CODE}/.git/config" | awk '{print $3}')
+  SCAN_NAME="${SCAN_NAME:-no_scan_name}"
```
Comment (david-piiano) [inline cli/flows-cli.sh:147]: Still recommended to split to:
1. Check CURL returns 200 OK - if not fail with message
2. Check that JSON contains what is expected

**Pull request merged**